sycl/include/sycl/ext/intel/esimd/memory.hpp

```diff
@@ -5862,9 +5862,9 @@ __ESIMD_API simd<T, N> slm_atomic_update_impl(simd<uint32_t, N> offsets,
 template <atomic_op Op, typename T, int N>
 __ESIMD_API std::enable_if_t<__ESIMD_DNS::get_num_args<Op>() == 0, simd<T, N>>
 slm_atomic_update(simd<uint32_t, N> byte_offset, simd_mask<N> mask = 1) {
-  // 2 byte, 8 byte types, non-power of two, and operations wider than 32 are
-  // supported only by LSC.
-  if constexpr (sizeof(T) == 2 || sizeof(T) == 8 ||
+  // 1 byte, 2 byte, 8 byte types, non-power of two, and operations wider than
```

Comment:
I doubt that 1-byte elements atomic_update is supported even with LSC.

Reply:
Can you please add some test cases to compile-only memory_properties.cpp (no cache-hints + word-vectors turn to LSC atomic_update) and E2E tests with word elements (maybe with byte-elements too if they work)?

Reply:
just pushed a commit with compile time tests. we should already have e2e tests, see here

[pred](https://github.com/intel/llvm/blob/sycl/sycl/test-e2e/ESIMD/unified_memory_api/Inputs/atomic_update.hpp#L639)

https://github.com/intel/llvm/blob/sycl/sycl/test-e2e/ESIMD/unified_memory_api/Inputs/atomic_update_slm.hpp#L567